tests/integration_tests/functional/test_drive_vhost_user.py

```diff
@@ -36,3 +97,295 @@ def test_vhost_user_block(microvm_factory, guest_kernel, rootfs_ubuntu_22):
+
+    # Converting path from tmpfs ("./srv/..") to local
+    # path on the host ("../build/..")
+    rootfs_path = utils.to_local_dir_path(str(rootfs_ubuntu_22))
+    # Launching vhost-user-block backend
+    _backend = spawn_vhost_user_backend(vm, rootfs_path, vhost_user_socket_1, True)
+
+    # We need to setup ssh keys manually because we did not specify rootfs
+    # in microvm_factory.build method
+    ssh_key = rootfs_ubuntu_22.with_suffix(".id_rsa")
+    vm.ssh_key = ssh_key
+    vm.spawn()
+    vm.basic_config(add_root_device=False)
+    vm.add_net_iface()
+
+    # Adding first block device.
+    fs1 = drive_tools.FilesystemFile(os.path.join(vm.fsfiles, "scratch1"), size=128)
+    vm.add_drive("scratch1", fs1.path)
+
+    # Adding second block device (rootfs)
+    vm.add_vhost_user_drive("rootfs", vhost_user_socket_1, is_root_device=True)
+
+    # Adding third block device.
+    fs2 = drive_tools.FilesystemFile(os.path.join(vm.fsfiles, "scratch2"), size=512)
+    vm.add_drive("scratch2", fs2.path)
+
+    # Launching vhost-user-block backend
+    _backend2 = spawn_vhost_user_backend(vm, rootfs_path, vhost_user_socket_2, False)
+    # Adding forth block device.
+    vm.add_vhost_user_drive("dummy_rootfs", vhost_user_socket_2)
+
+    vm.start()
+
+    # Determine the size of the microVM rootfs in bytes.
+    rc, stdout, stderr = utils.run_cmd(
+        "du --apparent-size --block-size=1 {}".format(rootfs_ubuntu_22),
+    )
+    assert rc == 0, f"Failed to get microVM rootfs size: {stderr}"
+
+    assert len(stdout.split()) == 2
+    rootfs_size = stdout.split("\t")[0]
+
+    # The devices were added in this order: fs1, rootfs, fs2. fs3
+    # However, the rootfs is the root device and goes first,
+    # so we expect to see this order: rootfs, fs1, fs2. fs3
+    # First check drives order by sizes.
+    ssh_connection = vm.ssh
+    _check_block_size(ssh_connection, "/dev/vda", rootfs_size)
+    _check_block_size(ssh_connection, "/dev/vdb", fs1.size())
+    _check_block_size(ssh_connection, "/dev/vdc", fs2.size())
+    _check_block_size(ssh_connection, "/dev/vdd", rootfs_size)
+
+    # Now check that vhost-user-block with rw is last.
+    # Prepare the input for doing the assertion
+    assert_dict = {}
+    # Keep an array of strings specifying the location where some string
+    # from the output is located.
+    # 1-0 means line 1, column 0.
+    keys_array = [
+        "1-0",
+        "1-6",
+        "2-0",
+        "2-6",
+        "3-0",
+        "3-6",
+        "4-0",
+        "4-6",
+    ]
+    # Keep a dictionary where the keys are the location and the values
+    # represent the input to assert against.
+    assert_dict[keys_array[0]] = "ro"
+    assert_dict[keys_array[1]] = "/dev/vda"
+    assert_dict[keys_array[2]] = "rw"
+    assert_dict[keys_array[3]] = "/dev/vdb"
+    assert_dict[keys_array[4]] = "rw"
+    assert_dict[keys_array[5]] = "/dev/vdc"
+    assert_dict[keys_array[6]] = "rw"
+    assert_dict[keys_array[7]] = "/dev/vdd"
```

Comment:
Can this be inlined in the dict declaration?

```
assert_dict = {
    "1-0": "ro",
...
```

It would be shorter and readable.

Reply:
done